lib/cretonne/build.rs

```diff
@@ -44,3 +46,55 @@ fn main() {
         process::exit(status.code().unwrap());
     }
 }
+
+fn output_target_cfgs() {
+    let config = env::var("CRETONNE_TARGETS").unwrap_or("all".into());
+    let possible_isas = ["riscv", "intel", "arm32", "arm64"];
+
+    match config.as_ref() {
+        "all" => {
+            for isa in &possible_isas {
+                println!("cargo:rustc-cfg=build_{}", isa);
+            }
+        }
+
+        "native" => {
+            let target = env::var("TARGET").expect("The TARGET environment variable must be set");
+            let arch = target.split("-").next().unwrap();
+
+            let isa = match arch {
+                "aarch64" => "arm64",
+                "arm" | "armv7" | "armv7s" => "arm32",
```

Comment:
Maybe we should follow [Clang's triple parser](http://www.llvm.org/docs/doxygen/html/Triple_8cpp_source.html#l00335) here? There's a number of ARM variants, so it just defers to `starts_with("arm")` (or thumb).